addons/mass_mailing_sms/models/mailing_mailing.py

```diff
@@ -324,6 +326,46 @@ def convert_links(self):
         res.update(super(Mailing, self - sms_mailings).convert_links())
         return res
 
+    def get_replacements_placeholders(self):
+        """Get placeholders for replaced links in sms widget for accurate computation of sms counts.
+
+        Reminders and assumptions:
+          * Links wille be transformed to the format "[base_url]/r/[link_tracker_code]/s/[sms_id]".
+          * unsubscribe is formatted as: "\nSTOP SMS : [base_url]/sms/[mailing_id]/[trace_code]".
+
+        :return: Character counts used for links, formatted as `{link: str, unsubscribe: str}`.
+        """
+        if self:
+            self.ensure_one()
+
+        self.check_access_rights('write')
+
+        query = '''
```

Comment:
mhe, still not happy with that, the length stuff should be really accurate with the search
the only use case is that if you remove manually some tracker code, and I don't think that complexity is worth to support that use case (espacially to display in a widget, nothing critical)

from a performance POV I'm not even sure it's "that better"

Reply:
OK. The other option for these values (not considering other undisputed changes)  was in https://github.com/odoo/odoo/commit/c7dea23a171adde28cbae8acb3f105364690c4bd.

The research part shows that changing it is indeed less readable.
And another contender with one caveat discussed here is https://github.com/odoo/odoo/pull/140102#discussion_r1450040539
@tde-banana-odoo , @std-odoo , let's somehow reach an agreement!

Reply:
Would say that I don't have strong opinion. As it is a tool method that does not really have any actionnable usage, for me it is ok to speedup the init of the widget with a query. If you feel it is not necessary, ok to go back to a search. both options are ok to me for this kind of tool method: sql is more "to the point", search is more "odoo style".

Reply:
It is sunny, I have confidence in Odoo, so would say to use a search finally, as sql should be the last defense when there are performance issues. So ok with @std-odoo